### PR TITLE
User Mail CC

### PR DESCRIPTION
**Pull request by metas-rc**

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5734590_A_User_CC.sql [hunk 1]
```diff
@@ -0,0 +1,91 @@
+-- 2024-09-24T19:11:58.811Z
+INSERT INTO AD_Element (AD_Client_ID,AD_Element_ID,AD_Org_ID,ColumnName,Created,CreatedBy,EntityType,IsActive,Name,PrintName,Updated,UpdatedBy) VALUES (0,583286,0,'CC_User_ID',TO_TIMESTAMP('2024-09-24 19:11:58.457000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',100,'D','Y','CC Nutzer/Kontakt','CC Nutzer/Kontakt',TO_TIMESTAMP('2024-09-24 19:11:58.457000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',100)
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5734590_A_User_CC.sql:2]: please add a description indicating that this uses is always getting a mail as Cc when the user in question receives a mail

**Review by metas-ts — APPROVED** [review]
i approved, but please add that field-descripton before integrating.  AD_User is broad in the functional sense. and this field might cause FUD, unless it'S clear that:
- it's just about email (not e.g. notification)
- it's always about **Cc** if the current user **receives** a mail

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/mail/MailRestController.java [hunk 2]
```diff
@@ -237,6 +243,7 @@ private WebuiEmail sendEmail(final WebuiEmail webuiEmail)
 			throw new FillMandatoryException("To");
 		}
 		final EMailAddress to = toList.get(0);
+		final EMailAddress cc = toList.size()>1 ? toList.get(1) : null;
```
Comment (teosarca) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/mail/MailRestController.java:246]: what happens with toList[2], toList[3] etc?

Comment (teosarca) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/mail/MailRestController.java:246]: checking at line 260, i see
```java
toList.stream().skip(1).forEach(email::addTo)
```

isn't there an overlap between `to`s and cc?

Comment (teosarca) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/mail/MailRestController.java:246]: I suggest to 
* either implement support for CC to webui mail (frontend too)
* either accept that Webui Mail does not support CC yet (is that in the requirements?)

**Review by teosarca — CHANGES_REQUESTED** [review]
pls check my comments

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/mail/MailRestController.java [hunk 3]
```diff
@@ -172,7 +172,13 @@ public JSONEmail createNewEmail(@RequestBody final JSONEmailRequest request)
 		final Integer toUserId = attributes.getAD_User_ID();
 		final LookupValue to = mailRepo.getToByUserId(toUserId);
 
-		final String emailId = mailRepo.createNewEmail(adUserId, from, to, contextDocumentPath).getEmailId();
+		final Integer ccUserId = attributes.getCC_User_ID();
+		final LookupValue cc = mailRepo.getToByUserId(ccUserId);
+
+		final List<LookupValue> lookupValues = ImmutableList.of(to, cc);
```
Comment (teosarca) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/mail/MailRestController.java:178]: NPE when cc is null?

**Review by teosarca — APPROVED** [review]

**Pull request merged**